src/gt4py/next/iterator/transforms/unroll_reduce.py

```diff
@@ -30,7 +30,14 @@ def _is_neighbors_or_lifted_and_neighbors(arg: itir.Expr) -> TypeGuard[itir.FunC
 
 
 def _get_neighbors_args(reduce_args: Iterable[itir.Expr]) -> Iterator[itir.FunCall]:
-    return filter(_is_neighbors_or_lifted_and_neighbors, reduce_args)
+    flat_reduce_args: list[itir.Expr] = []
```

Comment:
make sure there is a test for this case (I did not check when I was at this point)

Reply:
I've added a test `test_reduction_with_if`.